modules/annotation-processor/src/main/java/org/ballerinalang/natives/annotation/processor/BallerinaAnnotationProcessor.java

```diff
@@ -213,13 +186,44 @@ private void processNativeActions(Set<Element> balActionElements) {
             String className = Utils.getClassName(element);
             ActionHolder action = new ActionHolder(balAction, className);
             action.setAnnotations(getBallerinaAnnotations(element));
+            if (getPackage(packageName).getConnector(balAction.connectorName()) == null) {
+                ConnectorHolder connector = new ConnectorHolder(balAction.connectorName(),
+                        balAction.connectorArgs());
+                getPackage(packageName).addConnector(balAction.connectorName(), connector);
+            }
+            validateConnectorParams(getPackage(packageName).getConnector(balAction.connectorName()),
+                    balAction);
             getPackage(packageName).getConnector(balAction.connectorName()).addAction(action);
         }
     }
-    
+
+    private void validateConnectorParams(ConnectorHolder connectorHolder, BallerinaAction balAction) {
+        //need to validate connector params types as well as their relative order to be exactly same
+        //for all the actions who share the same ballerina connector
+        if (connectorHolder.getConnectorArgs().length > 0) {
+            if (connectorHolder.getConnectorArgs().length != balAction.connectorArgs().length) {
+                throw new BallerinaException("Connector Arguments should be equal and ordered the same across all"
+                        + "the ballerina actions registered against the same ballerina connector.");
+            }
+            int counter = 0;
+            for (; counter < connectorHolder.getConnectorArgs().length; counter++) {
+                Argument connectorArg = connectorHolder.getConnectorArgs()[counter];
+                Argument actionProvidedArg = balAction.connectorArgs()[counter];
+                if (connectorArg.name().equals(actionProvidedArg.name())
+                        && connectorArg.structType().equals(actionProvidedArg.structType())
+                        && (connectorArg.type().compareTo(actionProvidedArg.type()) == 0)
+                        && (connectorArg.elementType().compareTo(actionProvidedArg.elementType()) == 0)) {
```

Comment:
Used the enum.compareTo() here, however as you said == and equals should provide the same result.